app/gimlet/src/main.rs

```diff
@@ -68,47 +68,89 @@ fn system_init() {
     // you've flashed the wrong one, only.
     //
     // The revision is on pins PG[2:0], with PG2 as the MSB.
+    //
+    // In earlier firmware, we used the SP's internal pull-up resistors.
+    // However, the combination of both FPGA and SP pull-up resistors was too
+    // strong on certain hardware revisions, leading to marginal voltage
+    // readings.
+    //
+    // Instead, we reset the FPGA, which puts its pins into their default
+    // configuration (with weak pull-ups enabled).  Then, we wait for the pins
+    // to charge before reading the values.
 
-    // Un-gate the clock to GPIO bank G.
+    // Un-gate the clock to GPIO bank G (for revision ID), D (for FPGA reset),
+    // and C (for sequencer PG)
     p.RCC.ahb4enr.modify(|_, w| w.gpiogen().set_bit());
+    p.RCC.ahb4enr.modify(|_, w| w.gpioden().set_bit());
+    p.RCC.ahb4enr.modify(|_, w| w.gpiocen().set_bit());
```

Comment:
Out of curiosity: is it important that this be done in three separate writes? Or, could we do
```suggestion
    p.RCC.ahb4enr.modify(|_, w| {
        w.gpiogen().set_bit());
        w.gpioden().set_bit());
        w.gpiocen().set_bit());
    });
```
if having the writes be separate is important, it could be nice to write that down so someone doesn't come along and "refactor" it.

Reply:
I was initially unsure whether they were all on `ahb4enr`; consolidated!